src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs

```diff
@@ -11957,14 +12076,146 @@ public static class StaticType
     }
 }
 """;
-        var comp = CreateCompilation(src, targetFramework: TargetFramework.Net70, options: TestOptions.DebugExe);
+        var comp = CreateCompilation(src, targetFramework: TargetFramework.Net70);
         CompileAndVerify(comp, expectedOutput: IncludeExpectedOutput("ran"), verify: Verification.FailsPEVerify)
            .VerifyDiagnostics();
 
         var tree = comp.SyntaxTrees.First();
         var model = comp.GetSemanticModel(tree);
         var memberAccess = GetSyntax<MemberAccessExpressionSyntax>(tree, "C<int>.StaticType");
         Assert.Equal("E<System.Int32>.StaticType", model.GetSymbolInfo(memberAccess).Symbol.ToTestDisplayString());
+
+        var e = ((Compilation)comp).GlobalNamespace.GetTypeMember("E");
+        AssertEx.SetEqual(["E<T>.StaticType"], model.LookupSymbols(position: 0, e, name: "StaticType").ToTestDisplayStrings());
+        AssertEx.SetEqual(["E<T>.StaticType"], model.LookupSymbols(position: 0, e, name: "StaticType", includeReducedExtensionMethods: true).ToTestDisplayStrings());
+
+        var c = ((Compilation)comp).GlobalNamespace.GetTypeMember("C");
+        AssertEx.SetEqual([], model.LookupSymbols(position: 0, c, name: "StaticType").ToTestDisplayStrings());
+        AssertEx.SetEqual(["E<T>.StaticType"], model.LookupSymbols(position: 0, c, name: "StaticType", includeReducedExtensionMethods: true).ToTestDisplayStrings());
+
+        var cInt = model.GetTypeInfo(memberAccess.Expression).Type;
+        Assert.Equal("C<System.Int32>", cInt.ToTestDisplayString());
+        AssertEx.SetEqual([], model.LookupSymbols(position: 0, cInt, name: "StaticType").ToTestDisplayStrings());
+        AssertEx.SetEqual(["E<System.Int32>.StaticType"], model.LookupSymbols(position: 0, cInt, name: "StaticType", includeReducedExtensionMethods: true).ToTestDisplayStrings());
+    }
+
+    [Fact]
+    public void ExtensionMemberLookup_MatchingExtendedType_GenericType_GenericMember()
```

Comment:
>ExtensionMemberLookup_MatchingExtendedType_GenericType_GenericMember [](http://example.com/codeflow?start=16&length=68)

Are we testing similar scenario in type-only context?

Reply:
Add `ExtensionMemberLookup_MatchingExtendedType_GenericMember_TypeOnlyContext`